theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-toolbar-contribution.tsx

```diff
@@ -82,6 +84,18 @@ export class TraceViewerToolbarContribution implements TabBarToolbarContribution
                 signalManager().fireResetZoomSignal();
             }
         });
+        registry.registerCommand(
+            TraceViewerToolbarCommands.OPEN_TRACE, {
```

Comment:
It's necessary to register an additional command? It's not possible to just directly use the "OpenTraceCommand"? (This is a genuine question. I don't know how commands work.)

Reply:
Since we cannot add buttons to phosphor widget headers, we're creating a toolbar command (https://theia-ide.org/docs/commands_keybindings/) to create a custom action in the widget toolbar, similar to the way we add zoom-in & zoom-out buttons to the TabBar Widget when Resources view is opened 

The OPEN_TRACE command has the label and folder icon assigned to it, and we register the command using the registerCommand function where we specify that we want it visible only for the opened traces widget. The registerCommand function also specifies callback function (OpenTraceCommand) for when its triggered.